umbraco/Umbraco-CMS
# Always use text elements in RTE JSON output

### Prerequisites


### Description

This PR ensures that we always keep the pseudo `#text` nodes when parsing RTE content to JSON, even when the container node does not contain anything else than a `#text` node (i.e. paragraphs without children, or links containing only text).

### Testing this PR

1. Make sure the delivery API is configured with `"RichTextOutputAsJson": true`.
2. Create some rich text content. It should contain:
   - Paragraphs with only text
   - Paragraphs with text and links
   - Paragraphs with only links
   - ...be creative 😄 
3. Verify that every text is wrapped in a `#text` element in the delivery API output for the content.

### Sample JSON output

Given this RTE content:

![image](https://user-images.githubusercontent.com/7405322/237031310-b867410c-b49f-46dc-bc1e-d6cc9c73ea8d.png)

...the JSON output should now look something like this:

```json
{
    "tag": "#root",
    "attributes": {},
    "elements": [{
            "tag": "p",
            "attributes": {},
            "elements": [{
                    "text": "A paragraph containing only text.",
                    "tag": "#text"
                }
            ]
        }, {
            "tag": "p",
            "attributes": {},
            "elements": [{
                    "text": "A paragraph with ",
                    "tag": "#text"
                }, {
                    "tag": "a",
                    "attributes": {
                        "route": {
                            "path": "/articles/",
                            "startItem": {
                                "id": "8991e10f-f309-42f3-a48a-cdfd277117aa",
                                "path": "articles"
                            }
                        },
                        "title": "Articles",
                        "anchor": "#"
                    },
                    "elements": [{
                            "text": "a link",
                            "tag": "#text"
                        }
                    ]
                }, {
                    "text": " and some text.",
                    "tag": "#text"
                }
            ]
        }, {
            "tag": "p",
            "attributes": {},
            "elements": [{
                    "tag": "a",
                    "attributes": {
                        "route": {
                            "path": "/en-articles/",
                            "startItem": {
                                "id": "b160f6d7-8e61-4b84-8754-c040ac460b26",
                                "path": "localized-articles"
                            }
                        },
                        "title": "Localized Articles"
                    },
                    "elements": [{
                            "text": "A paragraph with only a link.",
                            "tag": "#text"
                        }
                    ]
                }
            ]
        }, {
            "tag": "p",
            "attributes": {},
            "elements": [{
                    "text": "A paragraph",
                    "tag": "#text"
                }, {
                    "tag": "br",
                    "attributes": {},
                    "elements": []
                }, {
                    "text": "with multiple",
                    "tag": "#text"
                }, {
                    "tag": "br",
                    "attributes": {},
                    "elements": []
                }, {
                    "text": "line breaks.",
                    "tag": "#text"
                }
            ]
        }, {
            "tag": "p",
            "attributes": {},
            "elements": []
        }
    ]
}
```
